addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/HeaderBasedSessionManagementMethodType.java

```diff
@@ -205,10 +209,33 @@ public void processMessageToMatchSession(HttpMessage message, WebSession session
                         "processMessageToMatchSession {} # headers {} ",
                         message.getRequestHeader().getURI(),
                         hbSession.getHeaders().size());
+
+                Map<String, String> trackedCookies =
+                        Arrays.asList(hbSession.getHttpState().getCookies()).stream()
+                                .collect(Collectors.toMap(c -> c.getName(), c -> c.getValue()));
+
+                List<HttpCookie> cookies = message.getRequestHeader().getHttpCookies();
                 for (Pair<String, String> header : hbSession.getHeaders()) {
+                    if (HttpHeader.COOKIE.equalsIgnoreCase(header.first)) {
+                        String[] kv = header.second.split("=");
+                        if (!trackedCookies.containsKey(kv[0])) {
+                            cookies.add(new HttpCookie(kv[0], kv[1]));
+                        } else {
+                            LOGGER.debug(
+                                    "processMessageToMatchSession {} ignoring tracked cookie {} ",
+                                    message.getRequestHeader().getURI(),
+                                    kv[0]);
+                        }
+                        continue;
+                    }
+
```

Comment:
Shouldn't this use CookieUtils or something instead of kinda re-inventing the wheel so to speak?

Reply:
Just the splitting and array handling. I agree it isn't complex code I just think we already have handling in commonlib, maybe I'm wrong and not digging enough

Reply:
The values returned from HttpHeaderBasedSession.getHeaders() are not strictly speaking headers, they are config values. If the first string is HttpHeader.COOKIE then the value should be key=value, not a full cookie header string.
We _could_ use CookieUtils to parse it, but I dont think thats strictly necessary or even strictly correct.